src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-form/configuration-form.component.ts

```diff
@@ -0,0 +1,267 @@
+    } else if (configOption.type === 'entity_addr_t') {
+      return [this.ipv4Ipv6Validator(configOption)];
+    } else if (configOption.type === 'uuid_d') {
+      return [this.uuidValidator(configOption)];
+    }
+  }
+
+  getStep(type: string, value: number): number | undefined {
+    const numberTypes = ['uint64_t', 'int64_t', 'size_t', 'secs'];
+
+    if (numberTypes.includes(type)) {
+      return 1;
+    }
+
+    if (type === 'double') {
+      if (value !== null) {
+        const stringVal = value.toString();
+        if (stringVal.indexOf('.') !== -1) {
+          // Value type double and contains decimal characters
+          const decimal = value.toString().split('.');
+          return Math.pow(10, -decimal[1].length);
+        }
+      }
+
+      return 0.1;
+    }
+
+    return undefined;
+  }
+
+  private numberValidator(configOption): ValidatorFn[] {
+    const validators = [];
+    if (configOption.max && configOption.max !== '') {
+      this.maxValue = configOption.max;
+      validators.push(Validators.max(configOption.max));
+    }
+    if (configOption.min && configOption.min !== '') {
+      this.minValue = configOption.min;
+      validators.push(Validators.min(configOption.min));
+    }
+    if (configOption.type === 'double') {
+      validators.push(Validators.pattern('^-?[0-9]+(.[0-9]+)?$'));
+    } else {
+      validators.push(Validators.pattern('^-?[0-9]+$'));
+    }
+
+    return validators;
+  }
+
+  private ipv4Ipv6Validator(configOption): ValidatorFn {
+    const ipv4Ipv6Rgx =
+      '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]).){3}([0-9]|[1-9][0-9]|1[0-9]{2}|' +
+      '2[0-4][0-9]|25[0-5])$|^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9]).)*([A-Za-z]|[A-Za-z]' +
+      '[A-Za-z0-9-]*[A-Za-z0-9])$|^s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|((' +
+      '[0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d' +
+      '|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|' +
+      '2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}' +
+      '(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]d|1dd|[1-9]?d)' +
+      '(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4})' +
+      '{1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|' +
+      '1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|' +
+      '((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d))' +
+      '{3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:' +
+      '((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(:(((:' +
+      '[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)' +
+      '(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)?s*$';
+    return Validators.pattern(ipv4Ipv6Rgx);
```

Comment:
What happens if the input string is empty? By default the validation should not fail in this case. Check the existing validators in CdValidators.

Reply:
do we really need to validate IPv4 and IPv6 addresses with a regular expression?

Reply:
I'm totally open for nicer alternatives.

Reply:
What about `[0-0a-fA-F.:\[\]]'?

Reply:
Okay, thanks, if that's enough.

Reply:
Is that enough? I don't know

Reply:
It's from a book: https://stackoverflow.com/questions/23483855/javascript-regex-to-validate-ipv4-and-ipv6-address-no-hostnames. But I do understand, if we don't want to trust it.
Sorry but I don't have a better solution here :/

Reply:
[ngx-ip](https://github.com/shlomiassaf/ngx-ip) looks interesting, too.

Reply:
Are you allowing the different [address representations](https://en.wikipedia.org/wiki/IPv6#Address_representation)?

Reply:
We agreed on using @sebastian-philipp regex and moving it to a more central place in today's stand-up.